aws/service.ts

```diff
@@ -275,7 +275,7 @@ function computeImage(
                 if (!registryId) {
                     throw new RunError("Expected registry ID to be defined during push");
                 }
-                const credentials = await aws.ecr.getCredentials({ registryId: registryId });
+                const credentials = await aws.ecr.getCredentials({ registryId: await (<any>registryId).promise() });
```

Comment:
This is the interesting part of this change, it deals with the breaking change for the shape of data sources arguments. We should feel bad about having to do the downcast from the output here, but it's the quickest way to get this stuff unblocked (and more or less what was happening before).

Reply:
Bummer that we have to do this.  I opened https://github.com/pulumi/pulumi-cloud/issues/544 to track cleaning this (and the larger scope of the interaction with the docker buildandpush support here).

Reply:
i'm looking at this now.

Reply:
This change is no longer necessary.